app/contacts/contact_ftp.py

```diff
@@ -0,0 +1,216 @@
+import json
+import os
+import re
+import asyncio
+import aioftp
+import sys
+
+from app.utility.base_world import BaseWorld
+
+
+class Contact(BaseWorld):
+    def __init__(self, services):
+        self.name = 'ftp'
+        self.description = 'Accept agent beacons through ftp'
+        self.contact_svc = services.get('contact_svc')
+        self.file_svc = services.get('file_svc')
+        self.logger = BaseWorld.create_logger('contact_ftp')
+        self.host = self.get_config('app.contact.ftp.host')
+        self.port = self.get_config('app.contact.ftp.port')
+        self.directory = '/' + self.get_config('app.contact.ftp.server.dir')
```

Comment:
why do you need to pass this slash in?

Reply:
It simplifies things. I need the slash when I am declaring the user directories, so instead of concatenating the string twice and changing the code in 178 & 194 to os join path, just add the slash at the begining.